datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -504,6 +533,136 @@ def create_aws_auth(config):
+    if not isinstance(reader_config, dict):
+        raise ConfigurationError('The `reader` settings of field `auth_token` must be a mapping')
+
+    writer_config = config['writer']
+    if not isinstance(writer_config, dict):
+        raise ConfigurationError('The `writer` settings of field `auth_token` must be a mapping')
+
+    reader_type = reader_config.pop('type', '')
+    if not isinstance(reader_type, str):
+        raise ConfigurationError('The reader `type` of field `auth_token` must be a string')
+    elif not reader_type:
+        raise ConfigurationError('The reader `type` of field `auth_token` is required')
+    elif reader_type not in AUTH_TOKEN_READERS:
+        raise ConfigurationError(
+            'Unknown `auth_token` reader type, must be one of: {}'.format(', '.join(sorted(AUTH_TOKEN_READERS)))
+        )
+
+    writer_type = writer_config.pop('type', '')
+    if not isinstance(writer_type, str):
+        raise ConfigurationError('The writer `type` of field `auth_token` must be a string')
+    elif not writer_type:
+        raise ConfigurationError('The writer `type` of field `auth_token` is required')
+    elif writer_type not in AUTH_TOKEN_WRITERS:
+        raise ConfigurationError(
+            'Unknown `auth_token` writer type, must be one of: {}'.format(', '.join(sorted(AUTH_TOKEN_WRITERS)))
+        )
+
+    reader = AUTH_TOKEN_READERS[reader_type](reader_config)
+    writer = AUTH_TOKEN_WRITERS[writer_type](writer_config)
+
+    return AuthTokenHandler(reader, writer)
+
+
+class AuthTokenHandler(object):
+    def __init__(self, reader, writer):
+        self.reader = reader
+        self.writer = writer
+
+    def poll(self, **request):
+        token = self.reader.read(**request)
+        if token is not None:
+            self.writer.write(token, **request)
+
+
+class AuthTokenFileReader(object):
+    def __init__(self, config):
+        self._path = config.get('path', '')
+        if not isinstance(self._path, str):
+            raise ConfigurationError('The `path` setting of `auth_token` reader must be a string')
+        elif not self._path:
+            raise ConfigurationError('The `path` setting of `auth_token` reader is required')
+
+        self._pattern = config.get('pattern')
+        if self._pattern is not None:
+            if not isinstance(self._pattern, str):
+                raise ConfigurationError('The `pattern` setting of `auth_token` reader must be a string')
+            else:
+                self._pattern = re.compile(self._pattern)
+                if self._pattern.groups != 1:
+                    raise ValueError(
+                        'The pattern `{}` setting of `auth_token` reader must define exactly one group'.format(
+                            self._pattern.pattern
+                        )
+                    )
+
+        # Cache all updates just in case
+        self._token = None
+
+    def read(self, **request):
+        if self._token is None or 'error' in request:
```

Comment:
```suggestion
        if self._token is None or 'force_refresh' in request:
```

I think using a boolean `force_refresh` is more clear/explicit. WDYT?

Reply:
I prefer letting the semantic be an error and the reader decide what to do (maybe no refresh)